engines/grim/textobject.cpp

```diff
@@ -278,16 +279,11 @@ void TextObject::setupText() {
 			nextLinePos = message.size();
 			cutLen = nextLinePos;
 		}
-		Common::String currentLine;
+		Common::String currentLine(message.c_str(), message.c_str() + nextLinePos);
```

Comment:
Needless copy for RTL (and the comment is now redundant). Suggestion:

```c++
		Common::String currentLine;
		if (g_grim->getGameLanguage() == Common::HE_ISR)
			currentLine = Common::convertBiDiString(currentLine, Common::kWindows1255);
		else
			currentLine = Common::String(message.c_str(), message.c_str() + nextLinePos);
```

Reply:
please note that coverBiDiString requires the created string (as first parameter), so the copy is needed

Reply:
Ah, right. Sorry for the noise :)